extensions/applicationinsights-debugplugin-js/src/DebugPlugin.ts

```diff
@@ -0,0 +1,252 @@
+/**
+ * DebugPlugin.ts
+ * @copyright Microsoft 2020
+ */
+
+import {
+  BaseTelemetryPlugin, IConfiguration, CoreUtils,
+  IAppInsightsCore, IPlugin, ITelemetryItem, IProcessTelemetryContext, _InternalLogMessage, LoggingSeverity, _InternalMessageId, getNavigator,
+  ITelemetryPluginChain, InstrumentFunc, IInstrumentHooksCallbacks, IInstrumentCallDetails, InstrumentFuncs
+} from '@microsoft/applicationinsights-core-js';
+import { LoggerElement } from './components/helpers';
+import { tempStyle, permStyle } from './components/styleNodeSrc';
+import { DebugBin, DebugBinParent } from './components/debugBins';
+
+
+export default class DebugPlugin extends BaseTelemetryPlugin {
+  /**
+   * the root element of the logger
+   */
+  public myRootElement: HTMLDivElement;
```

Comment:
Lets remove all of these from being public and instead lets, wrap them in a closure -- I think this would be a good use of the DynamicProto see dependencies extension for usage example.

And then as part of being hidden within the closure we change the names as suggested by Hector as we won't need the "my" prefix as they wont be visible.

Reply:
Fixed!